### DISCUSSION TITLE
MPRIS ID For Freetube ?

### DISCUSSION THREAD

**Discussion by ghost** [FreeTubeApp/FreeTube#4005] [General]
I want to seek some clarity on the mpris id used by freetube ?
Does it need chromium id for mpris or it can work with freetube.

This is needed for a recent snapcraft review question on whether its appropriate for freetube to use chromium id for mpris (snap variant used chromium as mpris id for freetube till now) .

So is there any particular project specific mpris id for freetube from flatpak i can see freetube is also valid, is this correct ?

Can freetube be used as mpris id instead of chromium.

**Comment by absidue** [comment]
Took a while to figure out what MPRIS is, as you didn't explain it or give any hint at what it is, I finally stumbled across a page on the arch wiki: https://wiki.archlinux.org/title/MPRIS

If you install FreeTube through one of the official install methods e.g. `.deb`, what MPRIS ID does it list for FreeTube when you run `playerctl --list-all`?

If it doesn't already say FreeTube through some Electron magic, I would recommend you follow up on this thread on the snapcraft forum, to see if they ever figured out how to set the ID in an Electron app: https://forum.snapcraft.io/t/electron-apps-and-mpris-slot/21408

As far as I can tell, if Electron/Chromium isn't already magically setting it to FreeTube, the only solution we would have would be to write a completely custom MPRIS implementation, which seems to be what Cider does (would be rather undesirable, especially considering that I had no clue what MPRIS even is just 45mins ago).

**Comment by ghost** [comment]
Hey there, i assumed mpris is quite common among media player hence i didn't explained it.
Regarding the output of playerctl .

```
chromium.instance9631
```

It mentions chromium.

Though you didn't clear one of my question on why the flatpak variant seeks freetube as MPRIS if no custom ID Exists ?

**Comment by absidue** [comment]
You are asking a bunch of mostly non-Linux users highly specific Linux questions. As a Linux user, you are in a much better position to figure this stuff out, all I can do is stab in the dark internet searches.

Your best bet is probably to follow up on that snap forum thread I mentioned and ask if they ever figured out how to customise the ID.

**Comment by ghost** [comment]
@absidue  I believe than it will be better to satisfy that freetube currently works with chromium only as a mpris tag for now, i don't want the snap to be delayed more its already running late, from the playerctl its already clear that freetube insists on using chromium as mpris id itself. In the past too my attempts to use other mpris id for the app have been futile, i believe this can be sorted out later as i & we sit on the basis of sorting out this mpris stuff it will only further delay the snap package when its evident that the devs don't themself are confident/sure on having their custom id for mpris.

Thanks nevertheless, i will still keep the snap rely on chromium for now as there is clearly no custom id for mpris with the freetube team as of now.

Edit: This issue should be kept open if the upstream agrees ,  i will follow the forum thread mentioned and see if in future we can get some stuff from there.

**Reply by absidue** [reply]
Just to make it clear, because you seem to be saying that it is our fault, no it isn't.
That just what it is hardcoded in as at compile time of the prebuilt Electron releases (they don't change it from the default value in the chromium source code). So unless you figure out a way to set a custom one at runtime, the only two possible ways of having a custom one are:

- Creating a custom Electron build with a patch, which is very undesirable, firstly as we don't want more on our plates and secondly because building those custom releases would take 5+ hours, if the build jobs on the Electron repo are anything to go by. I think we can both agree that 5+ hours for every Electron release and the extra maintenance burden, just for a custom MPRIS ID, is not worth it, especially as the existing one doesn't affect functionality.
- Figuring out how to turn off Electron's built-in MPRIS implementation and creating our own custom one. Which again sounds like way too much work for something that doesn't affect functionality. (the few projects that have done that, that I stumbled across, seem to have done it at a time when Electron/Chromium's built-in MPRIS support was lacking, so they had a much bigger reason to do it than we would have)

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Converted this to discussion because this does not belong in our issues

**Comment by ghost** [comment]
@absidue I never meant to say that its the devs fault what i said was that currently the project doesn't have its custom mpris id and the devs too are not aware about the project having its custom mpris id, thats it nothing less or more. 🙃

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Linking https://github.com/FreeTubeApp/FreeTube/issues/97 just in case

**Reply by absidue** [reply]
I think that issue can probably be closed now as Electron has MPRIS support built in these days.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Closing https://forum.snapcraft.io/t/mpris-request-for-freetube-snap/34855/6
